File: EY.UbbstractThinkers.ProjectManagementPortal.Server/Models/ProjectTask.cs
using System;

namespace EY.UbbstractThinkers.ProjectManagementPortal.Server.Models
{
    public class ProjectTask
    {
        public Guid Uid { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public TaskStatus Status { get; set; }
        public DateOnly StartDate { get; set; }
        public DateOnly? EndDate { get; set; }

        public Guid ProjectUid { get; set; }
        public Project Project { get; set; }

        public string ResourceId { get; set; }
    }
}

File: ey.ubbstractthinkers.projectmanagementportal.client/src/models/ProjectTask.ts
import type { Dayjs } from "dayjs";
import type { TaskStatus } from "./TaskStatus";
import dayjs from "dayjs";
import { Default } from "./TaskStatus";

export interface ProjectTask {
  uid?: string;
  name: string;
  description: string;
  status: TaskStatus;
  startDate: Dayjs | string;
  endDate: Dayjs | string | null;
  projectUid: string;
  resourceId: string | null;
}

export const DefaultTask: ProjectTask = {
  name: "",
  description: "",
  status: Default,
  startDate: dayjs(),
  endDate: null,
  projectUid: "",
  resourceId: null,
};
